src/app/clusters/icd-management-server/ICDManagementCluster.cpp

```diff
@@ -109,141 +88,293 @@ void ICDManagementFabricDelegate::OnFabricRemoved(const FabricTable & fabricTabl
 }
 #endif // CHIP_CONFIG_ENABLE_ICD_CIP
 
-ICDManagementAttributeAccess::ICDManagementAttributeAccess() :
-    AttributeAccessInterface(MakeOptional(kRootEndpointId), IcdManagement::Id)
-{}
-
-void ICDManagementAttributeAccess::Init(PersistentStorageDelegate & storage, Crypto::SymmetricKeystore * symmetricKeystore,
-                                        FabricTable & fabricTable, ICDConfigurationData & icdConfigurationData)
+ICDManagementCluster::ICDManagementCluster(EndpointId endpointId, PersistentStorageDelegate & storage,
+                                           chip::Crypto::SessionKeystore & symmetricKeystore, FabricTable & fabricTable,
+                                           ICDConfigurationData & icdConfigurationData, OptionalAttributeSet optionalAttributeSet,
+                                           BitMask<IcdManagement::OptionalCommands> aEnabledCommands,
+                                           BitMask<IcdManagement::UserActiveModeTriggerBitmap> aUserActiveModeTriggerBitmap,
+                                           CharSpan aUserActiveModeTriggerInstruction) :
+    DefaultServerCluster({ endpointId, IcdManagement::Id }),
+    mStorage(storage), mSymmetricKeystore(symmetricKeystore), mFabricTable(fabricTable),
+    mICDConfigurationData(icdConfigurationData), mOptionalAttributeSet(optionalAttributeSet), mEnabledCommands(aEnabledCommands),
```

Comment:
Followup: It's a little weird that we let people pass false for whether StayActiveRequest is enabled but set the LITS feature flag.

We need to either document in the header that that bit in aEnabledCommands is ignored when the feature flag is set, or have a Startup() that fails if the two things are set inconsistently (and in the latter case, just check mEnabledCommands in the Accepted/GeneratedCommands methods below, since it will be set correctly).

Followup because we need to decide which of the two options is better, and that shouldn't hold up this PR.

Reply:
Added TODO.